libraries/AP_Compass/AP_Compass_BMM350.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "AP_Compass_config.h"
+
+#if AP_COMPASS_BMM350_ENABLED
+
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+#include "AP_Compass.h"
+#include "AP_Compass_Backend.h"
+
+#define BMM350_I2C_ADDR_MIN 0x14
+#define BMM350_I2C_ADDR_MAX 0x17
+
+
+class AP_Compass_BMM350 : public AP_Compass_Backend
+{
+public:
+    static AP_Compass_Backend *probe(AP_HAL::OwnPtr<AP_HAL::Device> dev,
+                                     bool force_external,
+                                     enum Rotation rotation);
+
+    void read() override;
+
+    static constexpr const char *name = "BMM350";
+
+private:
+    AP_Compass_BMM350(AP_HAL::OwnPtr<AP_HAL::Device> dev,
+                       bool force_external,
+                       enum Rotation rotation);
+
+    AP_HAL::OwnPtr<AP_HAL::Device> _dev;
+
+    /**
+     * @brief BMM350 magnetometer dut offset coefficient structure
+     */
+    struct bmm350_vector4f
+    {
+        float x;    // x axis
+        float y;    // y axis
+        float z;    // z axis
+        float temp; // Temperature
+    };
+
+    /**
+     * @brief BMM350 magnetometer cross axis compensation structure
+     */
+    struct bmm350_cross_axis
```

Comment:
```suggestion
    struct cross_axis
```

bmm350 is redundant here as you're within the class.  Similarly elsewhere